src/OpenTelemetry.Instrumentation.SqlClient/Implementation/SqlClientDiagnosticListener.cs

```diff
@@ -56,6 +56,13 @@ public override void OnCustom(string name, Activity activity, object payload)
                 case SqlDataBeforeExecuteCommand:
                 case SqlMicrosoftBeforeExecuteCommand:
                     {
+                        _ = this.commandFetcher.TryFetch(payload, out var command);
```

Comment:
I'd suggest to do this and Filter *after* the StartActivity() call. In typical case where user does not have filter, we want to first call StartActivity(), which will run samplers, and decide if this activity should even be collected. If not, we don't need to pay the cost of "fetching" command from payload. This PR alters the flow by always having to pay the cost of "fetching" command.